server/src/main/java/org/opensearch/extensions/DiscoveryExtensionNode.java

```diff
@@ -82,6 +82,17 @@ public List<ExtensionDependency> getDependencies() {
         return dependencies;
     }
 
+    public boolean dependenciesContain(ExtensionDependency dependency) {
+        for (ExtensionDependency extensiondependency : this.dependencies) {
+            if (dependency.getUniqueId().equals(extensiondependency.getUniqueId())
+                && dependency.getVersion().equals(extensiondependency.getVersion())) {
```

Comment:
We don't need if condition here. Logical operator returns a boolean value itself.
```suggestion
                return dependency.getUniqueId().equals(extensiondependency.getUniqueId()) && (dependency.getVersion().equals(extensiondependency.getVersion()) 
```